.buildkite/publish/dra/init_dra_publishing.sh

```diff
@@ -0,0 +1,153 @@
+
+# ensure we have a directory and permissions for our dependency report to be created
+# this path is set in the actual config file in
+# https://github.com/elastic/infra/tree/master/cd/release/release-manager/project-configs
+export DEPENDENCIES_REPORTS_DIR="${RELEASE_DIR}/dist/cd/output" # TODO, change this path?
+mkdir -p "${DEPENDENCIES_REPORTS_DIR}"
+chmod -R a+rw "${DEPENDENCIES_REPORTS_DIR}"
+
+# function to set our vault creds for DRA publishing
+function setDraVaultCredentials() {
+  set +x
+  echo "-- setting DRA vault credentials --"
+
+  # need to keep original vault addr to reset it
+  # after we're done or else it won't be set
+  # if we call this fxn again
+  export ORIGINAL_VAULT_ADDR="${VAULT_ADDR:-}"
+
+  DRA_CREDS=$(vault kv get -field=data -format=json kv/ci-shared/release/dra-role)
+  if [[ "${DRA_CREDS:-}" == "" ]]; then
+    echo "Could not get DRA credentials from vault"
+    exit 2
+  fi
+
+  export VAULT_ADDR="$(echo $DRA_CREDS | jq -r '.vault_addr')"
+  if [[ "${VAULT_ADDR:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  export VAULT_ROLE_ID="$(echo $DRA_CREDS | jq -r '.role_id')"
+  if [[ "${VAULT_ROLE_ID:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  export VAULT_SECRET_ID="$(echo $DRA_CREDS | jq -r '.secret_id')"
+  if [[ "${VAULT_SECRET_ID:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  unset DRA_CREDS
+  set -x
+}
+
+# function to unset the vault credentials.
+function unsetDraVaultCredentials() {
+  set +x
+  unset VAULT_ROLE_ID
+  unset VAULT_SECRET_ID
+
+  # and reset our original vault address
+  export VAULT_ADDR="${ORIGINAL_VAULT_ADDR:-}"
+
+  set -x
+}
+
+# function to generate dependency report.
+function generateDependencyReport() {
+  echo "name,version,url,license,sourceURL" > $1 # TODO, do this right
+}
+
+# generate the dependency report and publish SNAPSHOT artifacts
+if [[ "${PUBLISH_SNAPSHOT:-}" == "true" ]]; then
+  # note that this name is set by infra and must be in the format specified
+  dependencyReportName="dependencies-${VERSION}-SNAPSHOT.csv";
+  echo "-------- Generating SNAPSHOT dependency report: ${dependencyReportName}"
+  generateDependencyReport $DEPENDENCIES_REPORTS_DIR/$dependencyReportName
+
+  echo "-------- Publishing SNAPSHOT DRA Artifacts"
+  cp $RELEASE_DIR/dist/elasticsearch_connectors-${VERSION}.zip $RELEASE_DIR/dist/dra-artifacts/connectors-${VERSION}-SNAPSHOT.zip
+  setDraVaultCredentials
+  export WORKFLOW="snapshot"
+
+  # TODO - when ready, remove the gate and ungate test vars
+  if [[ "${DRA_PUBLISH_GATE:-}" == "" || "${DRA_UNGATE_TEST:-}" == "true" ]]; then
+    source "${PROJECT_ROOT}/.buildkite/publish/dra/publish-daily-release-artifact.sh"
+  else
+    echo "DRA_PUBLISH_GATE is locked. This is where we would call 'source ${PROJECT_ROOT}/.buildkite/publish/dra/publish-daily-release-artifact.sh'"
```

Comment:
What does the gate actually mean? Can we change the message to be actionable (e.g. pass DRA_PUBLISH_GATE to unlock the publish or something)?

Reply:
I _suspect_ this can be removed but would like to see if @markjhoy confirms this

Reply:
Yeah - it can be removed... this was to ensure it could be run without the final publish as a test... having the gate there is good for testing (as it can be set in the environment during a manual Buildkite run) - but if you don't want this testability without publishing - remove it...